internal-services/catalog/create-advisory-task.yaml

```diff
@@ -58,19 +61,21 @@ spec:
               key: git_author_email
       script: |
           #!/usr/bin/env sh
-          set -ox pipefail
+          set -eo pipefail
 
-          # This should get overwritten if any command fails
-          echo -n "Success" > $(results.result.path)
-
-          failfunc() {
+          exitfunc() {
               local err=$1
               local line=$2
               local command="$3"
-              echo -n "$0: ERROR '$command' failed at line $line - exited with status $err" > $(results.result.path)
-              exit 0 # exit the script cleanly as there is no point in proceeding past an error
+              if [ $err -eq 0 ] ; then
+                  echo -n "Success" > $(results.result.path)
+              else
+                  echo -n "$0: ERROR '$command' failed at line $line - exited with status $err" > $(results.result.path)
+              fi
+              exit 0 # exit the script cleanly as there is no point in proceeding past an error or exit call
           }
-          trap 'failfunc $? $LINENO "$BASH_COMMAND"' ERR # the task should never fail with nonzero exit code
+          trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT # due to set -e, this catches all EXIT and ERR calls and the
+                                                          # task should never fail with nonzero exit code
```

Comment:
Nit: If the comment is too long to fit on the same line, I would suggest putting it above the command (and use as many lines as needed) - adding an empty line below with a comment at the end looks strange.

Reply:
great idea. I hate how it looks now :slightly_smiling_face:

Reply:
fixed

Reply:
Awesome, thanks :)